clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts

```diff
@@ -347,3 +346,43 @@ export async function waitForDiagnosticsChange(file: string | vscode.Uri) {
 
   return result;
 }
+
+export async function triggerCompletionsAndWaitForResults() {
+  return new Promise<vscode.CompletionList | undefined>((resolve, reject) => {
+    // Polling function to repeatedly check for suggestion visibility
+    const checkSuggestions = async () => {
+      // Get the active text editor
+      const editor = vscode.window.activeTextEditor;
+      if (!editor) {
+        reject("No active editor found");
+        return;
+      }
+
+      // Use the registered completion provider
+      const position = editor.selection.active;
+      const document = editor.document;
+
+      await vscode.commands.executeCommand(
+        "editor.action.triggerSuggest",
+        editor.document.uri,
+      );
+
+      // Trigger the completion provider manually
+      const completions =
+        await vscode.commands.executeCommand<vscode.CompletionList>(
+          "vscode.executeCompletionItemProvider",
+          document.uri,
+          position,
+        );
```

Comment:
These may still throw and leave the promise unresolved. I suggest something like
```
export async function triggerCompletionsAndWaitForResults() {
  while (true) {
    if (!editor) throw Error();
    await trigger;
    const completions =await executeCompletionItemProvider;
    if (completions && completions.items.length > 0) return completions;
    await sleep(100);
  }
}
```